python/src/fast.cpp

```diff
@@ -183,4 +188,102 @@ void init_fast(nb::module_& parent_module) {
         Returns:
           array: The quantized version of ``w``
       )pbdoc");
+
+  m.def(
+      "metal_kernel",
+      [](std::string name,
+         std::string source,
+         std::map<std::string, nb::handle> inputs_,
+         std::map<std::string, std::vector<int>> output_shapes,
+         std::map<std::string, Dtype> output_dtypes,
+         std::tuple<int, int, int> grid,
+         std::tuple<int, int, int> threadgroup,
+         std::optional<std::map<std::string, nb::handle>> template_args_,
+         bool ensure_row_contiguous,
+         bool verbose,
+         StreamOrDevice s) -> std::map<std::string, array> {
+        std::map<std::string, fast::TemplateArg> template_args;
+        if (template_args_) {
+          for (auto [name, value] : template_args_.value()) {
+            // Handle bool, int and dtype template args
+            if (PyBool_Check(value.ptr())) {
+              bool bool_val = nb::cast<bool>(value);
+              template_args.insert({name, bool_val});
+            } else if (PyLong_Check(value.ptr())) {
```

Comment:
Nit: `nb::isinstance<int>`

Reply:
Thanks! I spent while looking through the nanobind docs trying to figure out how to do this. This way is much better.